docs/proposals/improved-schema-for-services.md

```diff
@@ -0,0 +1,562 @@
+                "key": {
+                  "type": "string"
+                },
+                "name": {
+                  "type": "string"
+                },
+                "optional": {
+                  "type": "boolean"
+                }
+              }
+            },
+            "fieldRef": {
+              "type": "object",
+              "required": [
+                "fieldPath"
+              ],
+              "properties": {
+                "apiVersion": {
+                  "type": "string"
+                },
+                "fieldPath": {
+                  "type": "string"
+                }
+              }
+            },
+            "resourceFieldRef": {
+              "type": "object",
+              "required": [
+                "resource"
+              ],
+              "properties": {
+                "containerName": {
+                  "type": "string"
+                },
+                "divisor": {
+                  "type": "string"
+                },
+                "resource": {
+                  "type": "string"
+                }
+              }
+            },
+            "secretKeyRef": {
+              "type": "object",
+              "required": [
+                "key"
+              ],
+              "properties": {
+                "key": {
+                  "type": "string"
+                },
+                "name": {
+                  "type": "string"
+                },
+                "optional": {
+                  "type": "boolean"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "detectBindingResources": {
+    "type": "boolean"
+  },
+  "envVarPrefix": {
+    "type": "string"
+  },
+  "mountPathPrefix": {
+    "type": "string"
+  },
+  "services": {
+    "type": "array",
+    "items": {
+      "type": "object",
+      "required": [
+        "group",
+        "kind",
+        "version"
+      ],
+      "properties": {
+        "envVarPrefix": {
+          "type": "string"
+        },
+        "group": {
+          "type": "string"
+        },
+        "id": {
+          "type": "string"
+        },
+        "kind": {
+          "type": "string"
+        },
+        "name": {
+          "type": "string"
+        },
+        "namespace": {
+          "type": "string"
+        },
+        "version": {
+          "type": "string"
+        }
+      }
+    }
+  }
+}
+
+```
+
+Note - removed `description` fields to make the sample concise. 
+</details>
+
+
+- There are 5 top level fields in the metadata -
+    - `application`
+    - `customEnvVar`
+    - `detectBindingResources`
+    - `envVarPrefix`
+    - `mountPathPrefix`
+    - `services`
+
+### Conversion of openAPIV3Schema to json schema
+
+We after discussions with adapters team have decided to convert the openAPIV3Schema to json schema so that it would be easy to consume and hence we would be working on the converted json schema version to derive information from it.
```

Comment:
There is no openapiV3Schema to JsonSchema convertor library for Go. We should go with openapiV3Schema schema and skip any conversions.

Reply:
removed this section